qiskit/primitives/estimator.py

```diff
@@ -78,15 +102,28 @@ def _call(
             )
         sorted_observables = [self._observables[i] for i in observables]
         expectation_values = []
-        for circ, obs in zip(bound_circuits, sorted_observables):
+        for circ, obs, metadatum in zip(bound_circuits, sorted_observables, metadata):
             if circ.num_qubits != obs.num_qubits:
                 raise QiskitError(
                     f"The number of qubits of a circuit ({circ.num_qubits}) does not match "
                     f"the number of qubits of a observable ({obs.num_qubits})."
                 )
-            expectation_values.append(Statevector(circ).expectation_value(obs))
+            final_state = Statevector(circ)
+            expectation_value = final_state.expectation_value(obs)
+            if shots is None:
+                expectation_values.append(expectation_value)
+            else:
+                expectation_value = np.real_if_close(expectation_value)
+                sq_obs = (obs @ obs).simplify()
+                sq_exp_val = np.real_if_close(final_state.expectation_value(sq_obs))
+                variance = sq_exp_val - expectation_value**2
+                standard_deviation = np.sqrt(variance / shots)
```

Comment:
Is it necessary to divide by `shots`?
The documentation of [normal](https://numpy.org/doc/stable/reference/random/generated/numpy.random.normal.html) says as follows.
> where μ is the mean and σ the standard deviation. The square of the standard deviation, $σ^2$, is called the variance.

If we use σ/shots as standard deviation of normal, the standard error of the outcome might be $σ/shots^2$. I'm not 100% sure though.

Reply:
It seems to me that this is the only way to reflect the effect of the shots when sampling. This is because standard error is the standard deviation of the _mean_.

Reply:
I got it. You are right. We need to shrink the deviation depending on the number of shots.